internal/errorlint/errorlint.go

```diff
@@ -0,0 +1,140 @@
+// Copyright 2023 Outreach Corporation. All Rights Reserved.
+
+// Description: See package comment for this one file package.
+
+// Package errorlint contains the necessary logic for the error/log/trace linter.
+package errorlint
+
+import (
+	"go/ast"
+	"go/token"
+	"strconv"
+	"unicode"
+	"unicode/utf8"
+
+	"github.com/getoutreach/lintroller/internal/common"
+	"github.com/getoutreach/lintroller/internal/reporter"
+	"golang.org/x/tools/go/analysis"
+)
+
+// name defines the name for the errorlint linter.
+const name = "errorlint"
+
+// doc defines the help text for the error linter.
+const doc = `Ensures that each static message in error/trace/log/ is lowercase.
+
+A valid example is the following:
+
+	func foo() { errors.New("org not found in launchdarkly rule")}`
+
+// Analyzer exports the errorlint analyzer (linter).
+var Analyzer = analysis.Analyzer{
+	Name: name,
+	Doc:  doc,
+	Run:  errorlint,
+}
+
+// errorlint defines linter for error/trace/log messages
+func errorlint(_pass *analysis.Pass) (interface{}, error) {
+	// Ignore test packages.
+	if common.IsTestPackage(_pass) {
+		return nil, nil
+	}
+
+	// Wrap _pass with reporter.Pass to take nolint directives into account.
+	pass := reporter.NewPass(name, _pass, reporter.Warn())
+	for _, file := range pass.Files {
+		// Ignore generated files and test files.
+		if common.IsGenerated(file) || common.IsTestFile(pass.Pass, file) {
+			continue
+		}
+	}
+
+	for expr := range pass.TypesInfo.Types {
+		pkgName, isCleanString := lintMessageStrings(expr)
+		if !isCleanString {
+			pass.Reportf(expr.Pos(), "%s message should be lowercase and last char should not be one of \". : ! \\n\"", pkgName)
+		}
+	}
+
+	return nil, nil
+}
+
+// lintMessageStrings examines error/trace/log strings for capitalization and valid ending
+func lintMessageStrings(expr ast.Expr) (string, bool) {
+	call, ok := expr.(*ast.CallExpr)
+	if !ok {
+		return "", true
+	}
+
+	if !isDotInPkg(call.Fun, "errors", "New") && !isDotInPkg(call.Fun, "errors", "Wrap") &&
+		!isDotInPkg(call.Fun, "errors", "Wrapf") && !isDotInPkg(call.Fun, "log", "Warn") &&
```

Comment:
* `fmt.Errorf` (stdlib fmt package)
* `errors.Errorf`, `errors.WithMessage`, `errors.WithMessagef` (github.com/pkg/errors)

those should probs all be accounted for too

Reply:
I didn't discuss fmt package earlier, in that case should we add fmt.Sprintf too?

Reply:
No, just `fmt.Errorf` because its a standard way of forming errors. It returns an `error`

Reply:
added changes